impl/src/main/java/org/ehcache/UserManagedCacheBuilder.java

```diff
@@ -315,9 +376,31 @@ public T build() {
     return otherBuilder;
   }
 
-  public final UserManagedCacheBuilder<K, V, T> withCacheEvents(CacheEventDispatcher<K, V> cacheEventNotificationService) {
+  public final UserManagedCacheBuilder<K, V, T> withEventDispatcher(CacheEventDispatcher<K, V> eventNotifier) {
+    UserManagedCacheBuilder<K, V, T> otherBuilder = new UserManagedCacheBuilder<K, V, T>(this);
+    otherBuilder.eventNotifier = eventNotifier;
+    return otherBuilder;
+  }
+
+  public final UserManagedCacheBuilder<K, V, T> withOrderedEventExecutor(ExecutorService orderedExecutor) {
+    UserManagedCacheBuilder<K, V, T> otherBuilder = new UserManagedCacheBuilder<K, V, T>(this);
+    otherBuilder.orderedExecutor = orderedExecutor;
```

Comment:
The builder methods should take care of not allowing to get in an invalid state. Either by throwing early or by performing the necessary _overwrite_ when user specifies conflicting configuration.